### PR TITLE
Update SELF-ASSESMENT.md

### LINKED ISSUE #4978: [Feature]: SELF-ASSESMENT.md Contribution
### Requirement

As part of the Security Pals initiative with the [TAG Security](https://github.com/cncf/tag-security) group we're performing security assessments on CNCF Projects and we want to accurately represent your project and contribute to SELF-ASSESMENT.md.

### Problem

We currently have an [assessment drafted](https://github.com/cp-57/tag-security/blob/main/assessments/projects/jaeger/self-assessment.md) for Jaeger but would appreciate your feedback to verify the information we’ve collected and add any other security relevant information we can. We’re interested in connecting with any maintainers that want to review or contribute to our assessment.

### Proposal

We would like to contribute [our security assessment](https://github.com/cp-57/tag-security/blob/main/assessments/projects/jaeger/self-assessment.md) to your repository and contribute it to the collection of TAG Security Assessments as a testament to the security of Jaeger after getting your feedback.

### Open questions

Would it be possible to meet with or contact a maintainer for a quick review of our assessment?
Are there any Jaeger security resources we should reference or individuals we could contact to more holistically assess Jaeger?

### PR DESCRIPTION
**Pull request by SarahSMou**
Updated SELF-ASSESMENT.md by a TAG Security Group

## Which problem is this PR solving?
- Resolves #4978

## Description of the changes
- Updated SELF-ASSESMENT.md for review by project maintainers

## How was this change tested?
- This document was reviewed by a [TAG Security](https://github.com/cp-57/tag-security) Group

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: SELF-ASSESMENT.md [hunk 1]
```diff
@@ -21,12 +31,13 @@
...
+_A table at the top for quick reference information, later used for indexing._
 |   |  |
 | -- | -- |
-| Software | https://github.com/jaegertracing/jaeger/  |
-| Security Provider | No  |
-| Languages | Go |
-| SBOM | [Software bill of materials](https://github.com/jaegertracing/jaeger/releases/latest/download/jaeger-SBOM.spdx.json) |
+| **Software** | [A link to Jaeger’s repository.](https://github.com/jaegertracing/jaeger)  |
+| **Security Provider** | No, the main function of this project is to enable distributed tracing in an organization’s tech infrastructure. Security is not the primary objective.  |
+| **Languages** | <ul><li>Go</li><li>Shell</li><li>Makefile</li><li>Python</li><li>Jsonnet</li><li>Dockerfile</li></ul> |
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:39]: we do not use Jsonnet

File: SELF-ASSESMENT.md [hunk 2]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+
+A Tracing or OpenTelemetry’s SDK downloaded on the client or host is used to generate tracing data. An “instrumented application” (ex. OpenTelemetry API) creates spans when receiving a request and attaches context info (trace id, span id, and baggage). Only Ids and baggage are propagated. Other info sent to Jaeger backend asynchronously (Jaeger SDK)
+
+#### Deprecated [Jaeger Agent](https://github.com/jaegertracing/jaeger/issues/4739) (NOT REQUIRED)
+
+The Jaeger Agent is a deprecated network daemon used for listening for spans sent over UDP.
+
+#### Jaeger Collector
+
+The Jaeger Collector receives processes, validates, cleans up/ enriches and stores traces in some backend data storage [(see supported)](https://www.jaegertracing.io/docs/1.50/deployment).
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:97]:
```suggestion
The Jaeger Collector receives processes, validates, cleans up/ enriches and stores traces in some backend data storage [(see supported)](https://www.jaegertracing.io/docs/latest/deployment).
```

File: SELF-ASSESMENT.md [hunk 3]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+
+ II. Deprecated [Jaeger agent](https://github.com/jaegertracing/jaeger/issues/4739) (NOT REQUIRED)
+
+ III. The Jaeger Collector
+
+ IV. Jaeger Query
+
+ V. Jaeger Ingester
+
+<img width="726" alt="jaeger_1" src="https://github.com/cp-57/tag-security/assets/109479938/b4b36eea-dd4b-46e2-b8ea-3239eeecf21c">
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:83]: I prefer to refer to Architecture in the docs, not reference some random image

File: SELF-ASSESMENT.md [hunk 4]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+_How does your software fit into the cloud native ecosystem?  (e.g. Flibber is integrated with both Flocker and Noodles which covers virtualization for 80% of cloud users. So, our small number of "users" actually represents very wide usage across the ecosystem since every virtual instance uses
+Flibber encryption by default.)_
+
+#### OpenTelemetry Integration:
+
+OpenTelemetry can be used in place of the deprecated Jaeger Agent.
+
+* The OpenTelemetry SDK can be used to create trace data for Jaeger’s collector to collect and then store in a backend database.
+
+#### Istio Integration:
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:316]: Jaeger does not integrate with Istio, please remove

File: SELF-ASSESMENT.md [hunk 5]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+#### Cloud Native storage option:
+
+Jaeger is compatible with a number of storage backends that can be used to store and analyze traces in a cloud-native environment, including Cassandra, Elasticsearch, and Kafka.
+
+* Features and benefits that are specific to each storage backend include cost-effectiveness, scalability, and high availability.
+* Trace volume, retention needs, and infrastructure already in place are a few of the factors that must be taken into consideration while selecting a storage backend.
+
+#### Kubernetes Integration:
+
+Jaeger's Helm chart allows it to be deployed in Kubernetes.
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:332]: also jaeger-operator

File: SELF-ASSESMENT.md [hunk 6]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+* Additionally, you may make custom dashboards to keep an eye on your microservices in a cloud-native environment by utilizing Jaeger's interface with Grafana, a potent data visualization and monitoring tool.
+* Jaeger excels in distributed trace capture, while Prometheus focuses on time-series metrics for system monitoring. Together, they provide a comprehensive view of distributed system behavior and performance.
+
+## Security issue resolution
+
+#### Responsible Disclosures Process.
+
+_A outline of the project's responsible disclosures process should suspected security issues, incidents, or vulnerabilities be discovered both external and internal to the project. The outline should discuss communication methods/strategies._
+
+Vulnerabilities are shared with users and then publicly.
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:351]: there are no "users" who are separate from "public"

File: SELF-ASSESMENT.md [hunk 7]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+
+#### Vulnerability Response Process.
+
+_Who is responsible for responding to a report. What is the reporting process? How would you respond?_
+
+A report can be sent via email at jaeger-tracing@googlegroups.com, open an issue on GitHub or send a message to the #jaeger channel on the CNCF Slack encrypted with Jaeger’s public key. Responses can be sent directly via the method used to contact the maintainers by the maintainers.
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:357]: incorrect. Official process is in the docs, there is no need to duplicate it here https://www.jaegertracing.io/report-security-issue/

File: SELF-ASSESMENT.md [hunk 8]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+
+## Appendix
+
+### Known Issues Over Time.
+
+_List or summarize statistics of past vulnerabilities with links. If none have been reported, provide data, if any, about your track record in catching issues in code review or automated testing._
+
+#### Miscellaneous Issues
+
+* The Jaeger architecture relies on a central collector to receive and store trace reports from agents and microservices. The collector exposes HTTP endpoints like /api/traces for trace data submission, lacking authentication. This vulnerability opens the possibility of a Server-Side Request Forgery (SSRF) attack, enabling a compromised microservice to submit malicious trace data to the collector. Implementing token-based access enhances collector security by restricting entry to authenticated agents.
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:373]: Jaeger does not *always* expose HTTP endpoints, it's up to the user to configure them with TLS. And we support mTLS for auth.

File: SELF-ASSESMENT.md [hunk 9]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+
+### [CII Best Practices](https://www.coreinfrastructure.org/programs/best-practices-program/).
+
+_Best Practices. A brief discussion of where the project is at with respect to CII best practices and what it would need to achieve the badge._
+
+* The Jaeger project has achieved passing level criteria under Open Source Security Foundation Best Practices and is in the process of obtaining the silver badge.
+* Jaeger-client-js and Jaeger-client-javascript is in the process of obtaining passing level criteria.
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:380]: all Jaeger clients are deprecated, there are no plans to do anything with them.

File: SELF-ASSESMENT.md [hunk 10]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+### Case Studies.
+
+_Provide context for reviewers by detailing 2-3 scenarios of real-world use cases._
+
+#### Ticketmaster:
+
+* Consisting of over 300 microservices, the company benefits from the Jaeger project, utilizing the system to track over 100 million transactions a day. 
+* As the company scaled, they quickly learned that logging was insufficient and bulky. It helped uncomplicate the log aggregation process which generated terabytes of data daily. Jaeger simplified the debugging process by providing visibility into critical areas. – where developers could find root causes of issues and save time doing so.
+* Ticketmaster uses the Directed Acyclic Graph (DAG) view to see request flow. 
+* They also take advantage of Jaeger’s powerful adaptive sampling feature which reduces overhead without reducing visibility into systems.
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:391]: summary is fine, but why not link to the actual case study blog post?

File: SELF-ASSESMENT.md [hunk 11]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+
+#### Grafana Labs:
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:393]: @joe-elliott I think this was true when published, but is GL still using Jaeger?

File: SELF-ASSESMENT.md [hunk 12]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+
+* They needed a solution to trace request issues end-to-end. Grafana sometimes dealt with high request latency and opaque request paths. 
+* Jaeger allowed engineers to sort requests by duration with the Jaeger UI, which identified slow problematic requests and allowed for more granular troubleshooting. 
+* The immediate effect of this higher efficiency troubleshooting was the ability to increase query optimization, sometimes by 5 to 10x. 
+* One feature they particularly like is the contextual logging capability.
+
+#### Logz.io:
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:400]: @jkowall I don't know where this information is from, is it accurate?

File: SELF-ASSESMENT.md [hunk 13]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+
+### Related Projects / Vendors.
+
+_Reflect on times prospective users have asked about the differences between your project and projectX. Reviewers will have
+the same question._
+
+**ZipKin** was an earlier open source distributed tracing system which is used to help users monitor and troubleshoot microservice-based architectures. Both Zipkin and Jaeger aim to provide visibility into the flow of requests and responses across various services in a distributed system. While Zipkin has been around longer, Jaeger is known for its scalability to handle tracing in large and complex microservices and displaying those traces on the Web UI. Jaeger also has backward compatibility with Zipkin to help users transition from Zipkin to Jaeger. While Zipkin has been around longer than Jaeger, Jaeger has the benefit of being a part of Cloud Native Computing Foundation(CNCF), supporting containers, kubernetes and OpenTracing. To conclude, the decision between the two comes down to preference, supported languages and whatever is compatible with your existing tech stack.
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:412]:
```suggestion
**Zipkin** was an earlier open source distributed tracing system which is used to help users monitor and troubleshoot microservice-based architectures. Both Zipkin and Jaeger aim to provide visibility into the flow of requests and responses across various services in a distributed system. While Zipkin has been around longer, Jaeger is known for its scalability to handle tracing in large and complex microservices and displaying those traces on the Web UI. Jaeger also has backward compatibility with Zipkin to help users transition from Zipkin to Jaeger. While Zipkin has been around longer than Jaeger, Jaeger has the benefit of being a part of Cloud Native Computing Foundation(CNCF), supporting containers, kubernetes and OpenTracing. To conclude, the decision between the two comes down to preference, supported languages and whatever is compatible with your existing tech stack.
```

File: SELF-ASSESMENT.md [hunk 14]
```diff
@@ -44,3 +56,359 @@ See [README](https://github.com/jaegertracing/jaeger/#jaeger---a-distributed-tra
...
+* One feature they particularly like is the contextual logging capability.
+
+#### Logz.io:
+
+* The company was dealing with seemingly random decreasing performance issues in certain request flows. Logs failed to provide the answer or clue they were looking for. 
+* They incorporated Jaeger into their Node.js HTTP layers. 
+* The Jaeger UI immediately identified a series of issues including the serial call (a.k.a “staircase”) and a “span spike” (a request that ranged from a few milliseconds to 10 seconds).
+* They correlate logs with traces. These combined allow for a powerful troubleshooting system. Logz.io does so by attaching a request ID field to every log and adding it as a tag to the respective span.
+
+### Related Projects / Vendors.
```
Comment (yurishkuro) [inline SELF-ASSESMENT.md:407]: Please remove this. I don't want to maintain comparative analysis with competitors - things change all the time. I don't even want to maintain a list of possible competitors - why should we advertise them on our site?

File: SELF-ASSESMENT.md (hunk 12 above)
Comment (jkowall) [reply SELF-ASSESMENT.md:400]: @yurishkuro I do not recall where this came from and was unable to find anything on it. 

@SarahSMou I would suggest using the first two bullets and the last one. You could mention that the logz.io product offering includes Jaeger inside it.

File: SELF-ASSESMENT.md (hunk 10 above)
Comment (yurishkuro) [reply SELF-ASSESMENT.md:391]: https://medium.com/jaegertracing/ticketmaster-traces-100-million-transactions-per-day-with-jaeger-38ec6cf599f0

**Comment by jkowall** [comment]
@SarahSMou Can you address the requested changes here? Thanks for the contribution!

**Comment by eddie-knight** [comment]
:wave: Most of the feedback from this PR appears to have been applied by another contributor in a PR to the TAG Security repo. https://github.com/cncf/tag-security/pull/1198/files

As the TAG Security repo is the preferred location for this document, I suggest closing this PR and moving discussion to the new one.

If / when that PR becomes satisfactory, the self assessment on this repo could be removed and the security insights data can point to the new one instead.

**Pull request closed without merging**